core/store/src/lib.rs

```diff
@@ -404,6 +411,28 @@ impl StoreUpdate {
         }
         self.storage.write(self.transaction)
     }
+
+    #[cfg(feature = "protocol_feature_flat_state")]
+    pub fn apply_change_to_flat_state(&mut self, change: &RawStateChangesWithTrieKey) {
```

Comment:
StoreUpdate ideally shouldn't know about `FlatState`. Can we move this functionality to `shard_tries.rs`?